eth2util/keystore/keystore.go

```diff
@@ -121,6 +133,104 @@ func LoadKeys(dir string) ([]tbls.PrivateKey, error) {
 	return resp, nil
 }
 
+// LoadKeysSequential returns all secrets stored in dir/keystore-([0-9]*).json 2335 Keystore files
+// using password stored in dir/keystore-([0-9]*).txt.
+// The keystore files are read based on their index, and the returned slice is sorted accordingly.
+func LoadKeysSequential(dir string) ([]tbls.PrivateKey, error) {
+	return loadFiles(dir, "keystore-*.json", func(files []string) ([]string, error) {
+		newFiles, indices, err := orderByKeystoreNum(files)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(indices) == 0 {
+			return nil, errors.New("empty keystore indices")
+		}
+
+		if indices[0] != 0 {
+			return nil, errors.New("keystore indices must start from zero",
+				z.Int("first_index", indices[0]))
+		}
+
+		for sliceIdx, idx := range indices {
+			if sliceIdx == 0 {
+				continue
+			}
+
+			lastIdx := indices[sliceIdx-1] + 1
+
+			if lastIdx != idx {
+				return nil, errors.New("indices are non sequential",
+					z.Int("expected", lastIdx),
+					z.Int("got", idx))
+			}
+		}
+
+		return newFiles, nil
+	})
+}
+
+// LoadKeys returns all secrets stored in dir/keystore-*.json 2335 Keystore files
+// using password stored in dir/keystore-*.txt.
+func LoadKeys(dir string) ([]tbls.PrivateKey, error) {
```

Comment:
please also add tests for other non charon filenames, `keystore-foo.json, keystore-bar-1,json, keystore-bar-2.json, keystore-bar-10.json`, asserting returned order. This prevents us breaking the API inadvertantly.